server/schedulers/balance_region.go

```diff
@@ -165,3 +174,29 @@ func (s *balanceRegionScheduler) transferPeer(cluster schedule.Cluster, region *
 	balanceRegionCounter.WithLabelValues("move_peer", fmt.Sprintf("store%d-in", target.GetId())).Inc()
 	return schedule.CreateMovePeerOperator("balance-region", cluster, region, schedule.OpBalance, oldPeer.GetStoreId(), newPeer.GetStoreId(), newPeer.GetId())
 }
+
+// hasPotentialTarget is used to determine whether the specified sourceStore
+// cannot find a matching targetStore in the long term.
+// The main factor for judgment includes StoreState, DistinctScore, and
+// ResourceScore, while excludes factors such as ServerBusy, too many snapshot,
+// which may recover soon.
+func (s *balanceRegionScheduler) hasPotentialTarget(cluster schedule.Cluster, region *core.RegionInfo, source *core.StoreInfo, opInfluence schedule.OpInfluence) bool {
+	filters := []schedule.Filter{
+		schedule.NewExcludedFilter(nil, region.GetStoreIds()),
+		schedule.NewDistinctScoreFilter(cluster.GetLocationLabels(), cluster.GetRegionStores(region), source),
```

Comment:
Also `StorageThresholdFilter` is a long-term filter.

Reply:
Now the balance region scheduler does not consider storage threshold anymore because when stores are almost full it will be mainly based on the remaining space for scheduling.